daemon/mgr/snapshot.go

```diff
@@ -0,0 +1,159 @@
+package mgr
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/alibaba/pouch/ctrd"
+
+	"github.com/alibaba/pouch/pkg/errtypes"
+	"github.com/containerd/containerd/snapshots"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Snapshot contains the information about the snapshot.
+type Snapshot struct {
+	// Key is the key of the snapshot
+	Key string
+	// Kind is the kind of the snapshot (active, committed, view)
+	Kind snapshots.Kind
+	// Size is the size of the snapshot in bytes.
+	Size uint64
+	// Inodes is the number of inodes used by the snapshot
+	Inodes uint64
+	// Timestamp is latest update time (in nanoseconds) of the snapshot
+	// information.
+	Timestamp int64
+}
+
+// SnapshotStore stores all snapshots.
+type SnapshotStore struct {
+	lock      sync.RWMutex
+	snapshots map[string]Snapshot
+}
+
+// NewSnapshotStore create a new snapshot store.
+func NewSnapshotStore() *SnapshotStore {
+	return &SnapshotStore{snapshots: make(map[string]Snapshot)}
+}
+
+// Add a snapshot into the store.
+func (s *SnapshotStore) Add(sn Snapshot) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.snapshots[sn.Key] = sn
+}
+
+// Get returns the snapshot with specified key. Returns error if the
+// snapshot doesn't exist.
+func (s *SnapshotStore) Get(key string) (Snapshot, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	if sn, ok := s.snapshots[key]; ok {
+		return sn, nil
+	}
+	return Snapshot{}, errors.Wrap(errtypes.ErrNotfound, "snapshot")
+}
+
+// List lists all snapshots.
+func (s *SnapshotStore) List() []Snapshot {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	var snapshots []Snapshot
+	for _, sn := range s.snapshots {
+		snapshots = append(snapshots, sn)
+	}
+	return snapshots
+}
+
+// Delete deletes the snapshot with specified key.
+func (s *SnapshotStore) Delete(key string) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	delete(s.snapshots, key)
+}
+
+// SnapshotsSyncer syncs snapshot stats periodically.
+type SnapshotsSyncer struct {
+	store      *SnapshotStore
+	client     ctrd.APIClient
+	syncPeriod time.Duration
+}
+
+// NewSnapshotsSyncer creates a snapshot syncer.
+func NewSnapshotsSyncer(store *SnapshotStore, cli ctrd.APIClient, period time.Duration) *SnapshotsSyncer {
```

Comment:
`NewSnapshotsSyncer ` will not be exported, rename it to `newSnapshotsSyncer ` would be better.